src/main/java/com/nirima/jenkins/plugins/docker/DockerCloud.java

```diff
@@ -366,9 +373,6 @@ public boolean apply(@Nullable Container container) {
      *
      */
     private synchronized boolean addProvisionedSlave(String ami, int amiCap) throws Exception {
-        if( amiCap == 0 )
-            return true;
-
```

Comment:
I will wait for @magnayn to weigh in on #142. If this change turns out to be holding up this pull request, I'll move it to another PR (or someone could pull only the first two commits).